### ISSUE TITLE
[3.3.0RC2] IllegalArgumentException in FileSortOrderByName.sortCloudFiles

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android#3034]
GPlay console
```
java.lang.IllegalArgumentException: 
  at java.util.TimSort.mergeHi (TimSort.java:899)
  at java.util.TimSort.mergeAt (TimSort.java:516)
  at java.util.TimSort.mergeCollapse (TimSort.java:439)
  at java.util.TimSort.sort (TimSort.java:245)
  at java.util.Arrays.sort (Arrays.java:1498)
  at java.util.ArrayList.sort (ArrayList.java:1470)
  at java.util.Collections.sort (Collections.java:201)
  at com.owncloud.android.utils.FileSortOrderByName.sortCloudFiles (FileSortOrderByName.java:52)
  at com.owncloud.android.ui.adapter.OCFileListAdapter.setSortOrder (OCFileListAdapter.java:685)
  at com.owncloud.android.ui.fragment.OCFileListFragment.sortFiles (OCFileListFragment.java:1110)
  at com.owncloud.android.ui.activity.FileDisplayActivity.onSortingOrderChosen (FileDisplayActivity.java:1276)
  at com.owncloud.android.ui.dialog.SortingOrderDialogFragment$OnSortOrderClickListener.onClick (SortingOrderDialogFragment.java:180)
  at android.view.View.performClick (View.java:6294)
  at android.view.View$PerformClick.run (View.java:24770)
  at android.os.Handler.handleCallback (Handler.java:790)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6494)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:440)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:807)
```

@tobiasKaminsky I would guess it is the same BigDecimal/int issue

**Comment by AndyScherzinger** [comment]
not related to #2946 so there seems to be yet another name scenario where the sorting seems to fail :/

**Comment by tobiasKaminsky** [comment]
Hopefully this is fixed by latest PR regarding sort: #3143 
Closing for now.
